### DISCUSSION TITLE
Route nicht gefunden ?

### DISCUSSION THREAD

**Discussion by 1maxi2** [devemux86/cruiser#282] [Q&A] [answered]
Es passiert immer öfter beim setzen eines Shappingpunkt oder Ziel das die Meldung " Route nicht gefunden " kommt. Das passiert mit offline Planung mit brouter. 
Auch dauert die Berechnung oft mehr als 30 Sekunden. 
Ist das ein Problem wenn die Route lang ist ? ( ca.900km )
Ist das Telefon zu alt / langsam ?
Liegt ein Bedienungsfehler vor ?
<img src="https://github.com/devemux86/cruiser/assets/124886589/81bf42ab-a0c5-48b0-b766-4705db4e8035" width="50%"/>
Ich werde versuchen die Route per Mail zu schicken.

**Comment by devemux86** [comment]
Long routes need many waypoints to be calculated quickly with BRouter (offline).
Also be careful with the avoid options. What do you have activated?

Lange Routen brauchen viele Wegpunkte, um mit BRouter (offline) schnell berechnet zu werden.
Seien Sie auch vorsichtig mit den Vermeidungsoptionen. Was haben Sie aktiviert?

**Reply by 1maxi2** [reply]
Vermeidungen: Mautstrassen, Autobahn, Wohnstrassen.

**Comment by devemux86** [comment]
@1maxi2 

Have you downloaded all BRouter data for your region? (see image)
The phone speed is also important. This is a long route > 900 km, it takes > 25".
But if we add more shaping points, the route can be calculated in half the time.

Haben Sie alle BRouter-Daten für Ihre Region heruntergeladen? (siehe Bild)
Die Telefongeschwindigkeit ist ebenfalls wichtig. Dies ist eine lange Route > 900 km, sie dauert > 25".
Aber wenn wir mehr Shaping-Punkte hinzufügen, kann die Route in der Hälfte der Zeit berechnet werden.

<img src="https://github.com/devemux86/cruiser/assets/3484020/4ce875e2-16a0-4e56-b01b-cc18202014c0" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
BRouter calculates each route section separately. (like Cruiser rerouting!)
So BRouter speed is significantly improved if we set more waypoints on long routes.

BRouter berechnet jeden Streckenabschnitt separat. (wie Cruiser Rerouting!)
Die Geschwindigkeit des BRouters wird also erheblich verbessert, wenn wir auf langen Routen mehr Wegpunkte setzen.

**Reply by 1maxi2** [reply]
Ja Region ist heruntergeladen.
Ich habe versucht möglichst wenig Shappingpunkte zu verwenden da ich dachte je mehr Shappingpunkte desto länger dauert die Berechnung.
Ich werde jetzt beim nächsten mal mehr Punkte verwenden. 
Ich kann auch die Routenlänge verkürzen weil 900 km schaffe ich an 1 Tag sowieso nicht !
Mehr Sorgen macht mir die Meldung
 " Route nicht gefunden " Aber vielleicht gibt es dieses Problem mit mehr Punkten auch nicht mehr.
<img src="https://github.com/devemux86/cruiser/assets/124886589/5a3ba0d8-9c62-475e-9a12-7d93c8243cbe" width="50%"/>

**Reply by devemux86** [reply]
> Mehr Sorgen macht mir die Meldung " Route nicht gefunden "

This could be due to a timeout in the route calculation.

Dies könnte auf eine Zeitüberschreitung bei der Routenberechnung zurückzuführen sein.

> Aber vielleicht gibt es dieses Problem mit mehr Punkten auch nicht mehr.

:+1:

**Reply by airald** [reply]
I also see the message "route not found" often with BRouter. Sometimes is solved updating brouter but sometimes not. The time to compute the route is not related with this issue, because the message appears just in less than a second. 
Using Brouter (online) there's any problem. I always plan routes with a lot of waypoints (usually to indicate the names of close towns or give me extra informations like bypasses or other interesting data). 

I've been on route for 3 days and I found some roads closed. The new options to avoid closed roads or navigate to a waypoint on the list helped me a lot, redrawing a new route with a few taps.

**Reply by devemux86** [reply]
> Sometimes is solved updating brouter but sometimes not.

@airald do you mean updating the BRouter data with BRouter download manager in Cruiser?

If you see "Route not found" with _latest Cruiser_, please share some small route samples
and mention the BRouters (offline) avoid options, so I can check what happens.

It would be useful to make the message more explanatory when we discover the causes,
so that the users understand what is happening in these cases.

**Reply by airald** [reply]
updating brouter segments, using Cruiser or direct opening brouter app.

I've found that sometimes, when brouter seems is not able to find the route, restarting the phone solves the problem. I just test it now with a 100 meters route. Some minutes waiting for brouter with no success, restart the phone and founds inmedately.

**Comment by 1maxi2** [comment]
Ich habe jetzt wie von devemux empfohlen eine etwas kürzere Route geplant. ( Start in Senj) Dann habe ich in kürzeren Abständen das Ziel eingefügt. Dann immer wieder mit 
" Ziel erweitern " gearbeitet.
So konnte ich problemlos eine Tour auch auf Nebenstraßen planen ohne das die Fehlermeldung " Tour nicht gefunden " gekommen ist 😀
Das heißt für die Zukunft : Touren nicht allzu lange in einem Stück planen sonder lieber auf mehrere Touren aufteilen.
Lieber etwas mehr Shappingpunkte verwenden wo ich fälschlicherweise dachte das dies die Routenberechnung verlängert.

**Reply by devemux86** [reply]
@1maxi2 

Very nice process, step by step!
Cruiser also supports the "Extend route" function both at the end and at the beginning.

Sehr schöner Prozess, Schritt für Schritt!
Cruiser unterstützt auch die Funktion "Route verlängern" sowohl am Ende als auch am Anfang.

**Comment by Ciutnik** [comment]
I've been wondering for a long time whether a new feature like "Plan new route" would be useful. BEFORE the route is calculated, you could freely add, remove and move via points and/or shaping points. And only when everything is set on the map, clicking on "Calculate planned route" would start the calculation. 

This would be especially important for the offline BRouter, which, however, needs some time to calculate, and as we know, every change of point on the map causes a new calculation.

**Reply by devemux86** [reply]
@Ciutnik thanks, this an interesting suggestion.

The thing with this workflow, is that if we plan a larger route with several waypoints **without** immediate calculation
and the route cannot be calculated when we perform the calculation, we cannot know easily where the problem is.

While when we build a route step by step, it is easy to check which step or option may cause a wrong calculation.

BRouter does partial routing (like Cruiser navigation rerouting): it calculates routes segment by segment.
It is better to split the very large routes with shaping points. And start with small routes and extend them.

**Reply by Ciutnik** [reply]
Thank you very much for the detailed explanation!

Yes, I agree that my proposal may lead to a complete inability to calculate the route when we attempt a lot of waypoints, and at the same time we (or BRouter) make a "mistake" somewhere.
So, we put the idea aside 🙂

**Reply by 1maxi2** [reply]
Ich glaube auch das die Lösung besser ist wie sie jetzt ist.
So sehe sofort ob die Route zwischen den Shappingpunkten so verläuft wie ich mir das vorstelle. Wenn die Route fertig geplant ist ist das nachträgliche einfügen von Punkten meistens umständlicher.

**Comment by devemux86** [comment]
Cruiser now uses a maximum runtime of **10 minutes** for BRouter (offline) calculations.

(previous versions had a small maximum runtime of 1 minute)
